Ammunition/Shared/pellet.gd

```diff
@@ -1,11 +1,47 @@
 extends Node2D
+#defined externally
+@export var _pellet_speed: float = 0;
 
-@export var _pellet_speed: float = 15.0
-
+var stop_pos:Vector2;
+var ammo_base:AmmoType;
+var _start_position:Vector2;
 
+@warning_ignore("unassigned_variable")
+var unset_vec2 :Vector2;
+func _ready():
+	#bug that required this should be fixed
+	assert(global_position != unset_vec2)
+	
+	_start_position = global_position;
 func _physics_process(_delta):
-	global_position += Vector2.RIGHT.rotated(rotation) * _pellet_speed
-
-
+	var old_position := global_position;
+	#TODO find best distance to check
+	var position_delta := Vector2.RIGHT.rotated(rotation) * _pellet_speed;
+	var target := global_position + position_delta
```

Comment:
i'll split this into its own function, something like `_next_frame_position()`

Reply:
Unfortunately there was a bug that was in the visual display code which meant that the bullet speed was framerate dependent. Its fixed now but using the _previous_frame_position() will now require storing the previous frames delta value.